Fastfile.private

```diff
@@ -222,21 +222,29 @@ reflected in `Info.plist` during building."
 
     environment = options[:environment]
     build_configuration = get_build_configuration(environment: environment)
+    scheme = get_scheme(environment: environment)
+    project_file = Actions::ProjectNameAction.project_filename
 
     desc "Download provisioning profile and certificates if not present."
     refresh_certificates(
       environment: environment
     )
 
-    desc "Build the app using default project and scheme"
-    gym(
+    desc "Build the app using the project, build configuration and scheme"
+    params = {
+      scheme: scheme,
       configuration: build_configuration,
       include_symbols: true,
       # bitcode is disabled for the dsym file to keep valid after app is uploaded to app store
       # http://krausefx.com/blog/download-dsym-symbolication-files-from-itunes-connect-for-bitcode-ios-apps
       include_bitcode: false
-    )
-    ### Check if external_team_id needed
+    }
+    if project_file.include? ".xcodeproj" do
```

Comment:
There's always a ".xcodeproj" file. Better ask for `if project_file.include? ".xcworkspace" do` and do the opposite.

Reply:
Anyway, I'm not sure what you mean by doing this. Why is this necessary?

Reply:
Because in `gym` call, you must specify the project or the workspace with a different parameter (I guess that if you don't it infers it, but maybe it infers an incorrect one. In any case is better to make sure it uses the one set up in the configuration).
The action `project_file` gets the project file name according to configurations. So if the file name has a .xcodeproj, it is an xcodeproj, if not it's a workspace (or the other way around, it's the same)

Reply:
The point is that the `.xcodeproj` is always present, even if there's also a `.xcworkspace`. This is why I'd do the opposite.